backend/app/src/main/java/ch/zhaw/studyflow/domain/appointment/MemoryAppointmentDao.java

```diff
@@ -0,0 +1,49 @@
+package ch.zhaw.studyflow.domain.appointment;
+
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * In-memory implementation of the AppointmentDao interface.
+ */
+public class MemoryAppointmentDao implements AppointmentDao {
+    private final List<Appointment> appointments = new ArrayList<>();
+
+    @Override
+    public Appointment create(Appointment appointment) {
+        appointments.add(appointment);
```

Comment:
Maybe we should check for null here and set an id (auto-increment / simulate the database).